test/core/storage/ipfs/datastore_integration_test.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/ipfs/impl/datastore_leveldb.hpp"
+
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <libp2p/crypto/random_generator/boost_generator.hpp>
+#include "common/outcome_throw.hpp"
+#include "testutil/outcome.hpp"
+
+using fc::common::Buffer;
+using fc::storage::LevelDB;
+using fc::storage::ipfs::LeveldbDatastore;
+using libp2p::crypto::random::BoostRandomGenerator;
+using libp2p::crypto::random::CSPRNG;
+using libp2p::multi::ContentIdentifier;
+using libp2p::multi::HashType;
+using libp2p::multi::MulticodecType;
+using libp2p::multi::Multihash;
+
+struct DatastoreIntegrationTest : public ::testing::Test {
```

Comment:
I suggest you to reuse `test::BaseFS_Test`. It creates tmp dir and cleans up.

Reply:
Fixed temp dir creation and removal. I don't think I should use that base class, because it contains redundant data and profit is insignificant.